### PR TITLE
Improve consistency in functions across package (#72)

### PR DESCRIPTION
**Pull request by sdhutchins**
## Renamed functions

* `get_alpha_diversity_measures` to `alpha_diversity_measures`
  
## Added functions

* Added `stacked_barplots`, `alpha_diversity_plots`, `ordination_plots`
* Added `save_stacked_barplots`, `save_alpha_diversity_plots`, `save_ordination_plots`

## Deprecated functions

* Removed `save_barplot`

## Other
* Updated vignettes.

<hr>

Addresses #53 & #58.

### CODE REVIEW

**Comment by sdhutchins** [comment]
There may be a number of things that need changing. Please feel free to comb through this, @grabear.

**Comment by sdhutchins** [comment]
Something is causing the alpha diversity tests to fail. I'll check it later tonight. They're passing locally, but not travis.

File: vignettes/analysis.Rmd [hunk 1]
```diff
@@ -133,31 +157,43 @@ heat_tree_plots <- heat_tree_plots(metacoder_obj,
...
+# Generate stacked_barplots
+sb_plots <- stacked_barplots(metacoder_obj, tax_levels = c("Phylum", "Class", "Order"))
+names(sb_plots)
 ```
+
+
 ```{r message=FALSE, warning=FALSE, eval=FALSE}
 # Save plots with a custom output path
 save_heat_tree_plots(htrees = heat_tree_plots, custom_path = heat_tree_path)
 save_correlation_plots(corr = corr_plots, custom_path = corr_plot_path)
```
Comment (grabear) [inline vignettes/analysis.Rmd]:
```suggestion
save_correlation_plots(corr = corr_plots, custom_path = corr_plot_path)
save_stacked_barplots(sb_plots = sb_plots, custom_path = sb_plot_path)
```

File: vignettes/analysis.Rmd [hunk 2]
```diff
@@ -103,7 +107,7 @@ heat_tree_path <- output_dir(end_path = FALSE, start_path = "output", plot_type
 corr_plot_path <- output_dir(end_path = FALSE, start_path = "output", plot_type = "correlation")
```
Comment (grabear) [inline vignettes/analysis.Rmd]:
```suggestion
corr_plot_path <- output_dir(end_path = FALSE, start_path = "output", plot_type = "correlation")
sb_plot_path <- output_dir(end_path = FALSE, start_path = "output", plot_type = "stacked_barplot")
```

File: R/ordination.R [hunk 3]
```diff
@@ -96,11 +135,16 @@ ordination_plot <- function(obj, method = "PCoA", distance = "wunifrac", color =
...
 #' @importFrom crayon green
 #' @importFrom glue glue
-save_ordination_plots <- function(ord, format = "tiff", start_path = "output", ...) {
-  # Create the relative path to the ordination plots.  By default the path will be <pwd>/output/<experiment>/heat_trees/<format(Sys.time(), "%Y-%m-%d_%s")>
-  # With the parameters set the full path will be <pwd>/output/<experiment>/ord_plot/<extra_path>.
-  full_path <- output_dir(start_path = start_path, plot_type = "ord_plot", ...)
-  message(glue::glue(crayon::yellow("Saving Ordination Plots to the following directory: \n", "\r\t{full_path}")))
-
-  # ggplot2::ggsave(filename = paste0("ordination_", tolower(method), "_", tolower(distance), ".tiff"))
+save_ordination_plots <- function(plots, format = "tiff", start_path = "output", ...) {
```
Comment (grabear) [inline R/ordination.R]:
```suggestion
save_ordination_plots <- function(ord_plots, format = "tiff", start_path = "output", ...) {
```

File: R/ordination.R [hunk 4]
```diff
@@ -96,11 +135,16 @@ ordination_plot <- function(obj, method = "PCoA", distance = "wunifrac", color =
...
+  # Create the relative path to the ordination plots.  By default the path will be <pwd>/output/<experiment>/ordination/<format(Sys.time(), "%Y-%m-%d_%s")>
+  # With the parameters set the full path will be <pwd>/output/<experiment>/ordination/<extra_path>.
+  full_path <- output_dir(start_path = start_path, plot_type = "ordination", ...)
+  message(glue::glue(crayon::yellow("Saving Ordination plots to the following directory: \n", "\r\t{full_path}")))
+  # Iterate the plot list and save them in the proper directory
+  for (method in names(plots)) {
```
Comment (grabear) [inline R/ordination.R]:
```suggestion
  for (method in names(ord_plots)) {
```

File: R/ordination.R [hunk 5]
```diff
@@ -96,11 +135,16 @@ ordination_plot <- function(obj, method = "PCoA", distance = "wunifrac", color =
...
+    if (method != "metacoder_object") {
+      message(crayon::green("Saving the {method} ordination plot."))
+      ggplot2::ggsave(filename = sprintf("%s_ordination.%s", method, format), plot = plots[[method]], device = format, path = full_path, dpi = 500)
```
Comment (grabear) [inline R/ordination.R]:
```suggestion
      ggplot2::ggsave(filename = sprintf("%s_ordination.%s", method, format), plot = ord_plots[[method]], device = format, path = full_path, dpi = 500)
```

File: R/barplot.R [hunk 6]
```diff
@@ -109,33 +103,76 @@ stacked_barplot <- function(obj, tax_level = "Phylum", fill = "Phylum", xlabel =
...
+#'  \code{\link[MicrobiomeR]{output_dir}}
+#'
+#'  \code{\link[ggplot2]{ggsave}}
 #' @importFrom ggplot2 ggsave
-save_barplot <- function(plot, filename) {
-  if (is.null(filename)) {
-    filename <- plot
+#' @importFrom crayon yellow green
+#' @importFrom glue glue
+save_stacked_barplots <- function(plots, format = "tiff", start_path = "output", ...) {
```
Comment (grabear) [inline R/barplot.R]:
```suggestion
save_stacked_barplots <- function(sb_plots, format = "tiff", start_path = "output", ...) {
```

File: R/barplot.R [hunk 7]
```diff
@@ -109,33 +103,76 @@ stacked_barplot <- function(obj, tax_level = "Phylum", fill = "Phylum", xlabel =
...
+  # Create the relative path to the heat_tree plots.  By default the path will be <pwd>/output/<experiment>/heat_trees/<format(Sys.time(), "%Y-%m-%d_%s")>
+  # With the parameters set the full path will be <pwd>/output/<experiment>/heat_trees/<extra_path>.
+  full_path <- output_dir(start_path = start_path, plot_type = "stacked_barplots", ...)
+  message(glue::glue(crayon::yellow("Saving Stacked Barplots to the following directory: \n", "\r\t{full_path}")))
+  # Iterate the plot list and save them in the proper directory
+  for (rank in names(plots)) {
```
Comment (grabear) [inline R/barplot.R]:
```suggestion
  for (rank in names(sb_plots)) {
```

File: R/barplot.R [hunk 8]
```diff
@@ -109,33 +103,76 @@ stacked_barplot <- function(obj, tax_level = "Phylum", fill = "Phylum", xlabel =
...
+    if (rank != "metacoder_object") {
+      message(crayon::green("Saving the {rank} stacked barplot."))
+      ggplot2::ggsave(
+        filename = sprintf("%s_stacked_barplot.%s", tolower(rank), format), plot = plots[[rank]], device = format, path = full_path,
```
Comment (grabear) [inline R/barplot.R]:
```suggestion
        filename = sprintf("%s_stacked_barplot.%s", tolower(rank), format), plot = sb_plots[[rank]], device = format, path = full_path,
```

**Comment by sdhutchins** [comment]
> Shaurita, this looks great! We are almost done!
> 
> The only thing that I would request is this minor change in parameter name, and the following examples in the analysis vignette.

Awesome! I'm going to check this out and update the parameters in the documentation.

**Pull request merged**